kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSPropertyDeclarationImpl.kt

```diff
@@ -239,3 +233,11 @@ internal fun KaPropertySymbol.toModifiers(): Set<Modifier> {
     }
     return result
 }
+
+internal fun KSAnnotation.isValidOnProperty(): Boolean =
+    annotationType.resolve().declaration.annotations.none { metaAnnotation ->
+        metaAnnotation.annotationType.resolve().declaration.qualifiedName?.asString() == "kotlin.annotation.Target" &&
+            (metaAnnotation.arguments.singleOrNull()?.value as? ArrayList<*>)?.none {
```

Comment:
It was previously `any`, but now `none`. Is that expected?

Also, because of the inversion, now the util name sounds counter-intuitive. It is read like, if none of the annotation has non-Property target, it is valid on property?

Reply:
This is expected. The default behavior, i.e. without `@Target` on it, allows an annotation to be attached to a property. So an annotation is invalid on a property if and only if it has `@Target` that has no `PROPERTY`. This is equivalent to: an annotation is valid iff none of its meta annotations is "`@Target` without `PROPERTY`".